### ISSUE TITLE
Incremental Training Documentation

### ISSUE THREAD

**Issue by HannahKirk** [webis-de/small-text#9]
In `active_learner.py`, the incremental training parameter is described as:
```
incremental_training : bool
        If False, creates and trains a new classifier only before the first query,
        otherwise re-trains the existing classifier. Incremental training must be supported
        by the classifier provided by `clf_factory`."
```
Is there a way to retrain the model from scratch after each queried batch? This documentation suggests we are updating the existing classifier in both cases as even when `False`, it "creates and trains a new classifier _only_ before the __first__ query."

Thank you!

**Comment by chschroeder** [comment]
Thank your for this question, I have been waiting for this for some time now ;). This parameter is badly named and it has been bothering me for quite a while, but I could not decide for a better name. It also conflicts with a technical term in machine learning, if I remember correctly. In another project I have called a similar parameter `reuse_model_across_queries`, which describes its purpose a little better.

Even worse I seem to have screwed up the docstrings. The "otherwise" was meant to refer to "all queries except for the first one". With `incremental_training=False` the model is retrained from scratch after each queried batch.

This parameter is best left at its default value (`incremental_training=False`) if you want to be on the safe side. In some cases (`incremental_training=True`) works really well, but you have a little more risk here, since [fine-tuning transformers can be unstable](https://arxiv.org/pdf/2006.04884.pdf), especially when the training data consists of only a few examples. If you train from scratch you can recover from such a situation during the next iteration. (Disclaimer: The statement that it can work well is based on my own experiments performed on some selected configurations only.)

**Comment by HannahKirk** [comment]
Thank you for your reply! 
Strangely a few iterations, when I used `incremental_training=False` I get the following error message:
`Conflicting information about the number of classes: expected: 2, encountered: 1` even though the labeled indices definition have both class balances there (about 20\% negative and 80% positive).

But when I use `incremental_training = True`, there is no such error.

Could this be coming from the validation indices, not the train[labeled_indices]? I can't see why training from scratch on each iteration, the model is not observing both class labels when the labeled indices have both class labels.

Thanks :)

**Comment by chschroeder** [comment]
This should not be affected by `incremental_training` at all. You are right that this is probably caused by the validation set. How many samples do you have (and how many are labeled with 0/1 each?).

Thank you for your questions and feedback as well! These are very good and valid questions that help me improve the library, plus others can benefit from reading the issues here.

**Comment by chschroeder** [comment]
I forgot to include this in the respective commit message but this parameter is now called `reuse_model` and its docstring explanation is hopefully better formulated now. 

@everyone: Please tell me if this is not the case ;).
